### ISSUE TITLE
magento 2.2 unable to create configurable products and add video

### ISSUE THREAD

**Issue by msawyers** [magento/magento2#11706]
Preconditions

Previous version 2.2.0
PHP version 7.1.9

Steps to reproduce

Create Product
Add configurations 
Attempt to add video
Save  product

Expected Result

Product should save successfully
Conifugrable product should be created
video should be added

Actual result

Unable to add video to product
unable to create configurable products with associated simple products

**Comment by magento-engcom-team** [comment]
Hello @msawyers. Please confirm if You have tried to add video to the very configurable product (not to its concrete variations while creating configuration) by clicking on "Add Video" button in "Images And Videos" section of the product page.

**Comment by msawyers** [comment]
HI,

Yes I am confirming this. Selecting the add video button does nothing.

**Comment by msawyers** [comment]
Please note that you are unable to create configurable product as well, which is the main issue.

**Comment by magento-engcom-team** [comment]
Hi, @msawyers. Further we describe in detail the essential steps we have been undertaking while trying to reproduce that kind of behavior. Please specify in which of them the problem occurs in Your case. So:
1. Start to create configurable product.
2. Create configuration based on one or several attributes.
3. Expand the section "Images And Videos ".
4. Click on "Add Video" button. As a result new window called "New Video" appears veiling all the other things on the page.
5. In the "Url" text field enter the url of the video (from some video streaming site. for example "https://vimeo.com/117770305").
6. With the mouse click on  the other required field below, called "Title" or anywhere else on that window - video appears in the square (right-upper aria of the window) where till this time a video camera symbol has been displayed.
7. When all the fields marked with "*" have been filled, click "Save" in the left-upper corner of the window - the window with video information goes away and under the words, that are name of the section ("Images And Videos ") in a smaller square with photo camera symbol a picture of the uploaded video appears. Note, You couldn't have used this square (this icon) to add video. It is only for adding images. Nevertheless, when you have added the video its picture also appears in the same type of icon, just like it was an image.
8. Save the configurable product.

**Comment by msawyers** [comment]
Please see image attached.
![video](https://user-images.githubusercontent.com/8907629/32116828-c8c7a78a-bb11-11e7-99ea-0b77cab867b3.PNG)

**Comment by msawyers** [comment]
Additionally, when creating configurations and added associated products they disappear after save.

BEFORE

![before](https://user-images.githubusercontent.com/8907629/32117105-bec408f4-bb12-11e7-8b2f-d9e5eebda419.PNG)


AFTER

![after](https://user-images.githubusercontent.com/8907629/32117122-cf7d3332-bb12-11e7-940b-516477e4ee9f.PNG)

**Comment by magento-engcom-team** [comment]
Hi, @msawyers. Tell us please whether You are able to reproduce this issue on a fresh clear install of Magento 2.2.0 CE. Thank You.

**Comment by msawyers** [comment]
Okay, will update within 24hrs

**Comment by msawyers** [comment]
Hi, I am checking the differences between both environments. I see where default magento works fine.

**Comment by msawyers** [comment]
I believe the below error is related to issue of not being able to add videos? error happens when open add product option..

[2017-11-03 02:42:55] main.CRITICAL: The element 'gallery' already has a child with alias 'content' {"exception":"[object] (Magento\\Framework\\Exception\\LocalizedException(code: 0): The element 'gallery' already has a child with alias 'content' at /home/public_html/uat/dev/vendor/magento/framework/Data/Structure.php:611)"}

**Comment by magento-engcom-team** [comment]
Hello @msawyers. Tell us, please are there any  additional steps You do to the steps we've described here, and what do You exactly mean by "add product option".

**Comment by msawyers** [comment]
Hi, am now able to create configurable products. issue seem to have been due to using old static-content. however still not able to add videos. I am checking on what could have caused this issue,

**Comment by magento-engcom-team** [comment]
@msawyers, we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by driskell** [comment]
@msawyers Not sure if you ever resolved but this can happen if when upgrading Magento you copy files from new version to your installation, but do not remove deleted files from the target installation. In this case, between some specific versions of 2.1, the gallery block moved from one XML file to another and the original XML file was deleted - thus you get the error in your logs about duplication if the old file is still in the file tree. In our case the file we needed to delete was `app/code/Magento/ProductVideo/view/adminhtml/layout/catalog_product_form.xml` and we needed to scan for differences with existing Magento and repair.
